plugins/outputs/sql/sql.go

```diff
@@ -211,14 +277,11 @@ func (p *SQL) Write(metrics []telegraf.Metric) error {
 		tablename := metric.Name()
 
 		// create table if needed
-		if !p.tables[tablename] && !p.tableExists(tablename) {
-			createStmt := p.generateCreateTable(metric)
-			_, err := p.db.Exec(createStmt)
-			if err != nil {
+		if _, found := p.tables[tablename]; !found {
+			if err := p.createTableIfDoesNotExist(metric); err != nil {
```

Comment:
Aren't you generating the tables unconditionally now?

Reply:
Right, i was using "IF NOT EXIST" clause, which makes the subsequent query redundant. The only issue is that Microsoft SQL Server does not support it (every other driver does).

I thought about hard-coding it (by adding a workaround, when microsoft sql is in use), but the plugin starts to look like kinda bad, with driver-conditions in every other function.

I think using the tableExist function (for now) seems to be the most compatible approach.